### DISCUSSION TITLE
HtmlToFarHelp: выделение цветом и стилем

### DISCUSSION THREAD

**Discussion by johnd0e** [nightroman/FarNet#61] [Ideas]
Сейчас особенно не хватает возможности отличать
- разные уровни заголовков между собой (например заголовки первого уровня я бы фоном выделял)
- жирный текст от курсива

В идеале ещё бы возможность выделить цветом произвольный текст (span/div) с заданным атрибутом `class`.

**Comment by nightroman** [comment]
> жирный текст от курсива

Это должно быть возможно, но вашими усилиями с помощью вашей цветовой схемы.
Используйте свои цвета для регионов `markdown:StrongText`, `markdown:EmText`.

Так, например, я корректирую цвета других схем в своей цветовой схеме
https://github.com/nightroman/FarNet/blob/main/ColorerPack/hrd/visual.hrd

**Comment by johnd0e** [comment]
А в hlf?

**Comment by nightroman** [comment]
Пардон, думал, что это про маркдаун. Хотелось бы подробнее, что именно предлагается, примеры настроек. В частности, я не силен в HLF, как там можно задать цвет.

**Comment by johnd0e** [comment]
> ...как там можно задать цвет.

https://api.farmanager.com/ru/language/help_files.html#:~:text=%3C-,CtrlColorChar,-%3EXX

Примеры настроек:
```
ColorH1=70;
ColorH2=...;
...
ColorEmphasis=...;
ColorStrong=...;
ColorMark=...;
ColorKbd=...;
...
ColorClassSample1=...;
ColorClassSample2=...;
```

Последние два примера должны влиять на раскраску в случаях:
```html
<div class="Sample1">
Some DIV
<div>
<span class="Sample2">Some SPAN</span>
```

С `em` и `strong` объяснять думаю не надо, а элементы `mark`, `kbd`, `span` и `div` в pandoc markdown задаются особым синтаксисом, без html-тагов.

**Comment by nightroman** [comment]
Подумаю. Не обязательно возьмусь за это. Слишком много настроек - не обязательно хорошо. И мне не очень нравится, что цвета жесткие, это не очень хорошо может работать, если кто-то свою раскраску хелпа использует (я, например).

Вообще, по моей задумке изначальный дизайн такой, чтобы все хелпы были под одну гребенку. Это эргономично, после некоторой привычки. Не надо ментально переключаться на разные стили от разных авторов.

**Comment by johnd0e** [comment]
Идею в целом разделяю.

С другой стороны, я предлагаю немногим более чем возможность различать стандартные html-элементы, а стиля можно было бы придерживаться единого, определив например стандартные умолчательные значения.

Но отсутствие поддержки со стороны фара конечно проблема, на нестандартных схемах раскраски может понадобиться переборка hlf с другими значениями..

**Comment by nightroman** [comment]
@johnd0e Извините, не возьмусь за это. Не лежит душа, причины привел, и сейчас точно нет времени. Переведу в этот тикет в дискуссии, чтобы был более на виду, может, в будущем что-то поменяется по мере обсуждения.

**Comment by johnd0e** [comment]
> Вообще, по моей задумке изначальный дизайн такой, чтобы все хелпы были под одну гребенку.

> Слишком много настроек - не обязательно хорошо.

Хочу отметить, что я предлагаю вовсе не разнообразить оформление по прихоти автора.
Я предлагаю добавить расширить сам набор различаемых элементов.

Например сейчас, чтобы иметь возможность задать структуру заголовками H1,H2,H3 - приходится как-то разнообразить оформление, например заголовок может быть:
* центрированным, либо нет
* подсвеченным или нет
* отделённым горизонтальной чертой или нет
* и т.п.

В настоящий момент оформление зависит от прихоти автора, и отсюда неизбежный разнобой.
Вместо этого можно было бы определить стандартные стили (а давать ли сами настройки на откуп автору, это другой вопрос).

**Comment by johnd0e** [comment]
> И мне не очень нравится, что цвета жесткие, это не очень хорошо может работать, если кто-то свою раскраску хелпа использует.

Как я уже сказал, даже не трогая цвета, есть пространство для стилей.
А во-вторых, что если в фаре будет поддержка оформления в виде
```
\(::bold) Text \-
```
Где вместо bold может быть italic/inverse/underline/...
